packages/psss-server/src/connections/MeditrakConnection.ts

```diff
@@ -3,10 +3,112 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
+import { generateId } from '@tupaia/database';
+import { convertPeriodStringToDateRange } from '@tupaia/utils';
 import { ApiConnection } from './ApiConnection';
 
 const { MEDITRAK_API_URL = 'http://localhost:8090/v2' } = process.env;
 
+const SURVEY_RESPONSE_FIELDS = <const>['entity.code', 'survey.code', 'submission_time', 'id'];
+const ANSWER_FIELDS = <const>['question.code', 'id'];
+const PSSS_SURVEY_RESPONSE_ANSWER_TYPE = 'Number'; // All survey response answers are 'Number' in PSSS
+
+type SurveyResponseObject = {
+  [key in typeof SURVEY_RESPONSE_FIELDS[number]]: string;
+};
```

Comment:
Is this being too fancy? Basically I'm constructing a type here who's fields are defined in an array. It's pretty nifty but I dunno if it's necessary 🤷

Reply:
Yeah it is nifty but I'd prefer just listing them out in the object...just an extra thing to think about when you're working out the type. 

To confirm my understanding: `number` is a special keyword that TypeScript uses when mapping an array out into an object?

Reply:
>Yeah it is nifty but I'd prefer just listing them out in the object...just an extra thing to think about when you're working out the type.

Yeah, I think you're right.

>To confirm my understanding: number is a special keyword that TypeScript uses when mapping an array out into an object?

Yup, tbh this still feels a little like magic to me 🤷